ln-simln-jamming/src/parsing.rs

```diff
@@ -173,96 +176,196 @@ fn parse_duration(s: &str) -> Result<Duration, String> {
+        };
+        let path_clone = file_path.clone();
+        tasks.push(task::spawn(async move {
+            let mut file = File::open(path_clone)?;
+            file.seek(std::io::SeekFrom::Start(start))?;
+            let reader = BufReader::new(file).take(end - start);
+
+            let mut csv_reader = if i == 0 {
+                csv::Reader::from_reader(reader)
+            } else {
+                ReaderBuilder::new().has_headers(false).from_reader(reader)
+            };
+
+            let mut forwards = Vec::new();
+            for result in csv_reader.records() {
+                let record: StringRecord = result?;
+
+                // We can skip 6/7 because they're outgoing timestamps, we only care about when the htlc is fully removed
+                // from the incoming link (for simplicity's sake).
+                let incoming_amt: u64 = record[0].parse()?;
+                let incoming_expiry: u32 = record[1].parse()?;
+                let incoming_add_ts: u64 = record[2].parse()?;
+                let incoming_remove_ts: u64 = record[3].parse()?;
+                let outgoing_amt: u64 = record[4].parse()?;
+                let outgoing_expiry: u32 = record[5].parse()?;
+                let forwarding_node = PublicKey::from_slice(&hex::decode(&record[8])?)?;
+                let channel_in_id: u64 = record[10].parse()?;
+                let channel_out_id: u64 = record[11].parse()?;
+
+                // If we're filtering cut off any htlc that was in flight at the cutoff point.
+                if let Some(cutoff) = filter_cutoff {
+                    if incoming_add_ts > cutoff || incoming_remove_ts > cutoff {
+                        break;
+                    }
                 }
-            }
-            .add(duration.as_nanos() as u64);
 
-            if incoming_add_ts > cutoff || incoming_remove_ts > cutoff {
-                break;
+                let forward = BootstrapForward {
+                    incoming_amt,
+                    outgoing_amt,
+                    incoming_expiry,
+                    outgoing_expiry,
+                    added_ns: incoming_add_ts,
+                    settled_ns: incoming_remove_ts,
+                    forwarding_node,
+                    channel_in_id,
+                    channel_out_id,
+                };
+
+                forwards.push(forward);
             }
-        }
 
-        let forward = BootstrapForward {
-            incoming_amt,
-            outgoing_amt,
-            incoming_expiry,
-            outgoing_expiry,
-            added_ns: incoming_add_ts,
-            settled_ns: incoming_remove_ts,
-            forwarding_node,
-            channel_in_id,
-            channel_out_id,
-        };
+            Ok(forwards)
+        }));
+    }
 
-        forwards.push(forward);
+    let mut forwards = Vec::new();
+    for task in tasks {
+        let task_forwards = task.await??;
+        forwards.extend_from_slice(&task_forwards);
```

Comment:
We're no longer returning forwards in the order that they're written in the file, worth making a note of this in the function's doc (seems like the sort of thing you'd be surprised _doesn't_ happen)

Would also be nice to have a meta-note in the commit that we have to sort by add/remove ts _anyway_ later on so it's not worth the expense of sorting here.